web3/eth.py

```diff
@@ -403,6 +414,36 @@ async def get_transaction_count(
         mungers=[BaseEth.call_munger]
     )
 
+    async def get_transaction_receipt(
+        self, transaction_hash: _Hash32
+    ) -> TxReceipt:
+        return await self._get_transaction_receipt(transaction_hash)  # type: ignore
+
+    async def wait_for_transaction_receipt(
+        self, transaction_hash: _Hash32, timeout: float = 120, poll_latency: float = 0.1
+    ) -> TxReceipt:
+        try:
+            with Timeout(timeout) as _timeout:
+                while True:
+                    try:
+                        tx_receipt = await self._get_transaction_receipt(  # type: ignore
+                            transaction_hash
+                        )
+                    except TransactionNotFound:
+                        tx_receipt = None
+                    if tx_receipt is not None:
+                        break
+                    _timeout.sleep(poll_latency)
```

Comment:
Sorry to pop up again after approving this, but I think we want to be able to await this so it's not blocking. I think the most straightforward route is to get rid of the timeout context and use `await asyncio.sleep` here. If you want to see if it makes a difference at all, you could add it to our benchmarking suite and see what happens :)

Reply:
Makes sense. _timeout.sleep replaced with await asyncio.sleep